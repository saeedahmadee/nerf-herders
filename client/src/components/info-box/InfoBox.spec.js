import { shallowMount } from "@vue/test-utils";
import InfoBox from "./InfoBox.vue";

describe("InfoBox.vue", () => {
  const wrapper = shallowMount(InfoBox, {
    propsData: {
      info: {
        name: "TEST",
        description: "TEXT"
      }
    },
  });

  it('should render correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('Methods', () => {
    describe('closeModal', () => {
      it('should emit close when modal dismissed', () => {
        wrapper.vm.$refs.backdrop = 'test';
        wrapper.vm.closeModal({ target: 'test' });
        expect(wrapper.emitted().close).toBeTruthy();
      });
    });
  });
});
