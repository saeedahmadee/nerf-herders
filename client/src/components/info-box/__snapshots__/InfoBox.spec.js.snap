// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBox.vue should render correctly 1`] = `
<transition-stub
  name="modal"
>
  <div
    class="modal-mask"
  >
    <div
      class="modal-wrapper"
    >
      <div
        class="modal-container"
      >
        <div
          class="modal-head"
        >
          <span>
            TEST
          </span>
        </div>
         
        <div
          class="modal-body"
        >
          
            TEXT
           
          <button
            class="modal-default-button"
          >
            
            OK
          
          </button>
        </div>
      </div>
    </div>
  </div>
</transition-stub>
`;
